packages/react-paypal-js/src/v6/hooks/usePayPalGuestPaymentSession.ts

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { usePayPal } from "./usePayPal";
+import { useIsMountedRef } from "./useIsMounted";
+import { useError } from "./useError";
+import { useProxyProps } from "../utils";
+
+import type {
+    PayPalGuestOneTimePaymentSession,
+    PayPalGuestOneTimePaymentSessionOptions,
+    PayPalGuestOneTimePaymentSessionPromise,
+    PayPalGuestPresentationModeOptions,
+} from "@paypal/paypal-js/sdk-v6";
+import type { BasePaymentSessionReturn } from "../types";
+
+export type UsePayPalGuestPaymentSessionProps = (
+    | (Omit<PayPalGuestOneTimePaymentSessionOptions, "orderId"> & {
+          createOrder: () => PayPalGuestOneTimePaymentSessionPromise;
+          orderId?: never;
+      })
+    | (PayPalGuestOneTimePaymentSessionOptions & {
+          createOrder?: never;
+          orderId: string;
+      })
+) & {
+    buyerCountry?: string;
+    targetElement?: string | HTMLElement;
+    presentationMode?: "auto";
+    fullPageOverlay?: boolean;
+    autoRedirect?: boolean;
+};
+
+export function usePayPalGuestPaymentSession({
+    presentationMode = "auto",
+    fullPageOverlay,
+    createOrder,
+    orderId,
+    targetElement,
+    ...callbacks
+}: UsePayPalGuestPaymentSessionProps): BasePaymentSessionReturn {
+    const { sdkInstance } = usePayPal();
+    const isMountedRef = useIsMountedRef();
+    const sessionRef = useRef<PayPalGuestOneTimePaymentSession | null>(null);
+    const buttonRef = useRef<HTMLElement>(null);
+    const proxyCallbacks = useProxyProps(callbacks);
+    const [error, setError] = useError();
+    const [isProcessing, setIsProcessing] = useState(false);
```

Comment:
I don't see the `isProcessing` value used anywhere. What would this be used for?